src/SQLProvider.Runtime/Providers.MySql.fs

```diff
@@ -385,14 +385,26 @@ type internal MySqlProvider(resolutionPath, contextSchemaPath, owner:string, ref
             |> fun (x,_)-> x
             |> List.rev
             |> List.toArray
-            |> Array.unzip
+
+        let columnNames, values = columnNamesWithValues |> Array.unzip
 
         sb.Clear() |> ignore
-        ~~(sprintf "INSERT INTO %s (%s) VALUES (%s); SELECT LAST_INSERT_ID();"
+        ~~(sprintf "INSERT INTO %s (%s) VALUES (%s)"
             (entity.Table |> quotedTableName)
             ("`" + (String.Join("`, `",columnNames)) + "`")
             (String.Join(",",values |> Array.map(fun p -> p.ParameterName))))
 
+        match entity.OnConflict with
+        | Throw -> ()
+        | Update ->
+          ~~(sprintf " ON DUPLICATE KEY UPDATE %s"
+                (String.Join(",", columnNamesWithValues |> List.map(fun (c,p) -> sprintf "`%s`=%s" c p.ParameterName))))
+        | DoNothing ->
+          ~~(sprintf " AS x ON DUPLICATE KEY UPDATE %s"
+                (String.Join(",", columnNamesWithValues |> List.map(fun (c,p) -> sprintf "`%s`=x.`%s`" c p.ParameterName))))
+
+        ~~"; SELECT LAST_INSERT_ID();"
```

Comment:
From my testing on a MySQL instance this will return `0` if a row is updated instead of inserted. I tried to follow the code through to see where this value is used and my understanding is that an attempt is made to set the ID of the entity that was just created. But it looks like there is some code there to check whether the ID is already set, in which case this should be fine as we're only returning `0` in the case where it was an update and at that point it would have already existed anyway.

But please let me know if my understanding of that part of the code is wrong.